web3/utils/decorators.py

```diff
@@ -24,14 +25,15 @@ def reject_recursive_repeats(to_wrap):
 
     @functools.wraps(to_wrap)
     def wrapped(*args):
-        instances = tuple(map(id, args))
-        if instances in to_wrap.__already_called:
+        arg_instances = tuple(map(id, args))
+        unique_call = (threading.get_ident(),) + arg_instances
```

Comment:
I *think* it may be slighly easier to parse if you used a `threading.local` to store the state rather than keying based on the current thread.

Reply:
So I briefly considered that, but the problem that I ran into is I had to add my own namespace mechanism rather than just add the dict to the wrapped method, and check for/create a new dict if you're inside a new thread.

Maybe I missed an easier way to do it, but this seemed a lot cleaner. I can play with the variable names a bit to make it more readable, though.